File: ReactlikeMvvm/HiPadraoObservador/HiSujeitoBase.cs
using System.Collections.Generic;

namespace ReactlikeMvvm.HiPadraoObservador
{
    public class HiSujeitoBase<TArg>
    {
        private List<IHiObservador<TArg>> _observadores = new List<IHiObservador<TArg>>();
        public void Inscrever(IHiObservador<TArg> observador)
        {
            _observadores.Add(observador);
        }
        public void Desinscrever(IHiObservador<TArg> observador)
        {
            _observadores.Remove(observador);
        }
        public void NotificarTodos(TArg arg)
        {
            foreach (var iObservador in _observadores)
            {
                iObservador.Atualizar(arg);
            }
        }
    }
}

File: ReactlikeMvvmVb/HiPadraoObservador/HiSujeitoBase.vb
Imports System.Collections.Generic

Public Class HiSujeitoBase(Of TArg)
    Private Property _observadores As List(Of IHiObservador(Of TArg))
    Public Sub New()
        _observadores = New List(Of IHiObservador(Of TArg))()
    End Sub

    Public Sub Inscrever(observador As IHiObservador(Of TArg))
        _observadores.Add(observador)
    End Sub

    Public Sub Desinscrever(observador As IHiObservador(Of TArg))
        _observadores.Remove(observador)
    End Sub

    Public Sub NotificarTodos(arg As TArg)
        For Each iObservador As IHiObservador(Of TArg) In _observadores
            iObservador.Atualizar(arg)
        Next
    End Sub
End Class